test/alternator/test_scan.py

```diff
@@ -414,29 +414,38 @@ def test_scan_long_partition_tombstone_string(dynamodb, query_tombstone_page_lim
 # single returned page is still limited. DynamoDB specifies it should be
 # limited to 1 MB. In Alternator the limit is close to 1 MB, but it turns
 # out (see issue #10327) that for small tables the page size can grow up
-# to 3 MB. The following test accepts this as ok. Note that for larger tables,
-# the page size goes back to being closer to 1 MB.
+# signifficantly. The following test accepts this as ok. Note that for larger
+# tables, the page size goes back to being closer to 1 MB.
 #
 # This test is for Scan paging on a table with many small partitions. We have
 # a separate test for a Query over a single long partition with many rows -
 # test_query.py::test_query_reverse_longish (the test's name suggests it
 # checks reverse queries, but it also checks the unreversed unlimited query).
 # For single-partition scans, the page size is more exactly 1 MB.
-def test_scan_paging_missing_limit(dynamodb):
+#
+# The test is run twice, with two different values of N (the number of items in a batch write).
+# The value of 200 corresponds to 2000 KB of data in a batch write and it was shown to fail
+# with 4 or more initial tablets per shard due to #10327. In general, we want the paging limit
+# to be lower than 2000 KB, but we do not enforce it yet.
+# The value of 1400 (which translates onto 14000 KB of data put into a batch write)
+# was proven to be working even with 16 tablets per shard. We require it not to fail with this
+# value, so to know if paging works at all.
+@pytest.mark.parametrize("N", [
+    pytest.param(200, marks=pytest.mark.xfail(
+        reason="Issue #10327: small tables don't respect paging limits")),
+    1400
```

Comment:
I don't think we need to run the the test with the smaller number. The higher number already proves the "missing limit" works correctly (doesn't scan the entire table).
I am not happy we need to increase 600 to 1400 - it can make the test almost 3 times slower. I'll check myself if a lower number works reasonably and if it does, I'll lower it. I don't see any reason why we need to support 16 tablets per shard and not a different number.